### PR TITLE
Remove outdated information from contributing file

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
With https://github.com/mattermost/mattermost-developer-documentation/pull/247 getting merged, all information that was previously only available via this file, is now on https://developers.mattermost.com. Hence, I propose to remove most of this file and just link to https://developers.mattermost.com.

After/If this PR has been merged, I will submit PR's to update:
- https://github.com/mattermost/mattermost-webapp/blob/master/CONTRIBUTING.md
- https://github.com/mattermost/mattermost-mobile/blob/master/CONTRIBUTING.md
- https://github.com/mattermost/mattermost-redux/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by hanzei** [comment]
@jasonblais After you approved this PR (hope this happens :wink:), I will queue this for dev review.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,75 +1,3 @@
...
-   - Dev calls out on the issue that it is a major change and it will be merged after branching.
-   - Once the current release is branched the PR can be merged into `master`.
-
-#### PR Merged
-
-After a PR is merged:
-- External Contributions: PM closes the [Help Wanted] issue and related JIRA ticket.
-- Internal Contributions: Core committer resolves the JIRA ticket.
-- PM follows up for docs and changelog, and QA for release tests.
+Thank you for your interest in contributing! Please see the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/) which describes the process for making code contributions across Mattermost projects and [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) to ask questions from community members and the Mattermost core team.
```
Comment (jasonblais) [inline CONTRIBUTING.md]: Propose adding 

```
When you submit a pull request, it goes through a [code review process outlined here](https://developers.mattermost.com/contribute/getting-started/code-review/). We aim to start reviewing pull requests in this repo the week they are submitted, but the length of time to complete the process will vary depending on the pull request.
```

in a second paragraph.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,75 +1,5 @@
...
-   - Dev calls out on the issue that it is a major change and it will be merged after branching.
-   - Once the current release is branched the PR can be merged into `master`.
-
-#### PR Merged
-
-After a PR is merged:
-- External Contributions: PM closes the [Help Wanted] issue and related JIRA ticket.
-- Internal Contributions: Core committer resolves the JIRA ticket.
-- PM follows up for docs and changelog, and QA for release tests.
+When you submit a pull request, it goes through a [code review process outlined here](https://developers.mattermost.com/contribute/getting-started/code-review/). We aim to start reviewing pull requests the week they are submitted, but the length of time to complete the process will vary depending on the pull request.
```
Comment (lieut-data) [inline CONTRIBUTING.md]: nit: "The week they are submitted" implies we try to review things submitted on Friday the same day. Might suggest simplifying to the following, and letting the code-review documentation be authoritative:

```suggestion
When you submit a pull request, it goes through a [code review process outlined here](https://developers.mattermost.com/contribute/getting-started/code-review/). The length of time to complete the process will vary depending on the pull request.
```

Comment (hanzei) [reply CONTRIBUTING.md]: I would like to give contributors a rough idea when they can expect a review. What do you think about?

```suggestion
When you submit a pull request, it goes through a [code review process outlined here](https://developers.mattermost.com/contribute/getting-started/code-review/). We aim to start reviewing pull requests within a week, but the length of time to complete the process will vary depending on the pull request.
```

Comment (lieut-data) [reply CONTRIBUTING.md]: Should we encourage a contributor to stay on this page and not read about our code review process? I'm wondering if we'd be better off leaving them without this information to force the click-through.

Comment (hanzei) [reply CONTRIBUTING.md]: I like that reasoning. But using this force, I would remove the information about the review length entirely.
```suggestion
When you submit a pull request, it goes through a [code review process outlined here](https://developers.mattermost.com/contribute/getting-started/code-review/).
```

Comment (lieut-data) [reply CONTRIBUTING.md]: I'm in favour of that 👍

**Pull request merged**